### PR TITLE
add room header when externals are allowed

### LINKED ISSUE #186: Roomlist displays private and public rooms in the same way
This is confusing for users. Both forums and private rooms are in the same section of roomlist, and with the same icon.

### PR DESCRIPTION
**Pull request by odelcroi**
With the header : 
Capture d’écran 2022-10-18 à 16 30 24

Test : 
- [x] 1 : create a room with external allowed, check that the header is displayed
- [x] 2 : create a private room without external allowed, check that the header is NOT displayed
- [x] 3 : create a private room, after the creation, allow external in the settings, check that the header is displayed

e2e : 
- [x] 1 : create a room with external allowed, check that the header is displayed
- [x] 2 : create a private room without external allowed, check that the header is NOT displayed

UI : 
- [ ] : check responsiveness -> N/A

### CODE REVIEW

File: src/components/views/rooms/TchapExternalRoomHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { _t } from "matrix-react-sdk/src/languageHandler";
+
+import { TchapRoomAccessRule, TchapRoomAccessRulesEventId } from '../../../@types/tchap';
+import TchapRoomUtils from '../../../util/TchapRoomUtils';
+
+export interface IProps {
+    room?: Room;
+}
+
+export function checkExternalAllowed(room: Room): boolean {
```
Comment (estellecomment) [inline src/components/views/rooms/TchapExternalRoomHeader.tsx:13]: je mettrais bien cette fonction dans TchapRoomUtils

File: src/components/views/rooms/TchapExternalRoomHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    //listen to events to refresh the value if needed
+    useTypedEventEmitter(room.currentState, RoomStateEvent.Events, (ev: MatrixEvent) => {
+        console.log(ev.getType());
+        if (ev.getType() === TchapRoomAccessRulesEventId) {
+            setExternalAllowed(checkExternalAllowed(room));
+        }
+    });
+    //refresh the state after rendering
+    useEffect(() => {
```
Comment (estellecomment) [inline src/components/views/rooms/TchapExternalRoomHeader.tsx]: Hmmm.
I think the react lifecycle is : state change -> rendering -> effect.
Here the effect contains a state change (setExternalAllowed). Does that make an infinite loop ? state change -> rendering -> effect -> state change -> ...

I think that the rendering already uses isExternalAllowed to render the appropriate UI, so the effect is not necessary.

Comment (estellecomment) [reply src/components/views/rooms/TchapExternalRoomHeader.tsx]: I tried removing it, the status is still updated when you change externs status !

**Comment by estellecomment** [comment]
cypress is broken locally and in CI, but I can't figure out why. Later.

File: src/components/views/rooms/TchapExternalRoomHeader.tsx (hunk 1 above)
Comment (odelcroi) [reply src/components/views/rooms/TchapExternalRoomHeader.tsx:13]: si tu veux :)

File: src/components/views/rooms/TchapExternalRoomHeader.tsx (hunk 2 above)
Comment (odelcroi) [reply src/components/views/rooms/TchapExternalRoomHeader.tsx]: nice

Comment (odelcroi) [reply src/components/views/rooms/TchapExternalRoomHeader.tsx]: > Does that make an infinite loop ? state change -> rendering -> effect -> state change -> ...
the effect is trigger only if room is updated  as mentioned in the comment

Comment (odelcroi) [reply src/components/views/rooms/TchapExternalRoomHeader.tsx]: ok to remove it, it works well without it :)

**Pull request merged**